internal/services/sql/sql_managed_instance_failover_group_resource.go

```diff
@@ -0,0 +1,408 @@
+package sql
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/sql/mgmt/v5.0/sql"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/mssql/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/sql/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceSqlInstanceFailoverGroup() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceSqlInstanceFailoverGroupCreateUpdate,
+		Read:   resourceSqlInstanceFailoverGroupRead,
+		Update: resourceSqlInstanceFailoverGroupCreateUpdate,
+		Delete: resourceSqlInstanceFailoverGroupDelete,
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.InstanceFailoverGroupID(id)
+			return err
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.ValidateMsSqlFailoverGroupName,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"managed_instance_name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.ValidateMsSqlServerName,
+			},
+
+			"partner_managed_instances": {
+				Type:     pluginsdk.TypeList,
+				Required: true,
+				MaxItems: 1,
```

Comment:
Since we can only have one of these, should we simply this down to `partner_managed_instance`? I'd go that route unless support for more than one instance is coming

Reply:
Changed to `partner_managed_instance_id`